cmd/common/registry.go

```diff
@@ -0,0 +1,342 @@
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/common/frontends/console"
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	columnhelpers "github.com/inspektor-gadget/inspektor-gadget/internal/column-helpers"
+	gadgetrunner "github.com/inspektor-gadget/inspektor-gadget/internal/gadget-runner"
+	"github.com/inspektor-gadget/inspektor-gadget/internal/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/internal/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/internal/runtime"
+	cols "github.com/inspektor-gadget/inspektor-gadget/pkg/columns"
+	gadgetregistry "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-registry"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+// AddCommandsFromRegistry adds all gadgets known by the registry as cobra commands as a subcommand to their categories
+func AddCommandsFromRegistry(rootCmd *cobra.Command, runtime runtime.Runtime, columnFilters []cols.ColumnFilter) {
+	runtimeParams := runtime.Params()
+
+	// Build lookup
+	lookup := make(map[string]*cobra.Command)
+	for _, cmd := range rootCmd.Commands() {
+		lookup[cmd.Name()] = cmd
+	}
+
+	// Add runtime flags
+	addFlags(rootCmd, runtimeParams)
+
+	// Add operator global flags
+	operatorsParamsCollection := operators.OperatorsParamsCollection()
+	for _, operatorParams := range operatorsParamsCollection {
+		addFlags(rootCmd, operatorParams)
+	}
+
+	// Add all known gadgets to cobra in their respective categories
+	categories := gadgets.GetCategories()
+	for _, gadget := range gadgetregistry.GetGadgets() {
+		cmd, ok := lookup[gadget.Category()]
+		if !ok {
+			// Category not found, add it
+			categoryDescription, ok := categories[gadget.Category()]
+			if !ok {
+				panic(fmt.Errorf("category unknown: %q", gadget.Category()))
+			}
+			cmd = &cobra.Command{
+				Use:   gadget.Category(),
+				Short: categoryDescription,
+			}
+			rootCmd.AddCommand(cmd)
+			lookup[gadget.Category()] = cmd
+		}
+		cmd.AddCommand(buildCommandFromGadget(gadget, columnFilters, runtime, runtimeParams, operatorsParamsCollection))
+	}
+}
+
+func buildGadgetDoc(gadget gadgets.Gadget, columns columnhelpers.Columns) string {
+	var out strings.Builder
+	out.WriteString(gadget.Description() + "\n\n")
+
+	if columns != nil {
+		out.WriteString("Available columns:\n")
+		for columnName, description := range columns.GetColumnNamesAndDescription() {
+			out.WriteString("\t" + columnName + "\n")
+			if description != "" {
+				out.WriteString("\t  " + description + "\n")
+			}
+		}
+	}
+	return out.String()
+}
+
+func buildCommandFromGadget(gadget gadgets.Gadget, columnFilters []cols.ColumnFilter, runtime runtime.Runtime, runtimeParams params.Params, operatorsParamCollection params.ParamsCollection) *cobra.Command {
```

Comment:
```suggestion
func buildCommandFromGadget(gadget gadgets.Gadget, columnFilters []cols.ColumnFilter, runtime runtime.Runtime, runtimeParams params.Params, operatorsParamsCollection params.ParamsCollection) *cobra.Command {
```

BTW, please break this line.